### ISSUE TITLE
Self only rules do not work, but code looks like they should: Error: One of ['cidr_blocks', 'ipv6_cidr_blocks', 'self', 'source_security_group_id', 'prefix_list_ids'] must be set to create an AWS Security Group Rule

### ISSUE THREAD

**Issue by ossareh** [terraform-aws-modules/terraform-aws-security-group#168]
Hi there,

This is a really great module, good work!

I tend to use self only rules when configuring resources and their consumers. However while using this module I've found that causes an error through the code looks like it should work.

Given this module invocation:

```
module "redis_sg" {
  source  = "terraform-aws-modules/security-group/aws//modules/redis"
  version = "~> 3.0"

  name = "example_redis_sg"
  vpc_id = "vpcid-123123"
}
```

I get:

```
Error: One of ['cidr_blocks', 'ipv6_cidr_blocks', 'self', 'source_security_group_id', 'prefix_list_ids'] must be set to create an AWS Security Group Rule

  on .terraform/modules/scratch_nfs_access/terraform-aws-security-group-3.5.0/main.tf line 56, in resource "aws_security_group_rule" "ingress_rules":
  56: resource "aws_security_group_rule" "ingress_rules" {
```

It appears to happen with all the modules I've tried (ssh, redis, nfs).

The pattern employed in the module seems fine: https://github.com/terraform-aws-modules/terraform-aws-security-group/blob/master/modules/redis/main.tf#L15

I've tested this in terraform console and it ends up with an empty list `[]`. Which should result in length returning 0 here: https://github.com/terraform-aws-modules/terraform-aws-security-group/blob/master/main.tf#L56-L70

But it is that resource that is causing the error since there are none of the required parameters (`cidr_blocks`, `ipv6_cidr_blocks`, `self`, `source_security_group_id`, or `prefix_list_ids`) being passed.

Am I missing something basic? My work around has been to include `ingress_cidr_blocks = ["127.0.0.1/32"]` in the module invocation.

**Comment by antonbabenko** [comment]
Hi @ossareh !

I think you can specify extra argument like this:

```
ingress_with_self = [
    {
      rule = "redis-tcp"
    }
]
```

But I also think that there is no better solution than the one you are doing already.

Maybe you can find something more in `examples/complete`?

**Comment by ossareh** [comment]
Hi @antonbabenko, thanks for the suggestion - I tried that and the result is the same; it trips over the same issue I documented, i.e. the `count = length()` call in `main.tf` should be returning 0 but it is returning something other than 0 and resulting in that block being created without valid arguments.

For clarity the versions at play here are:

```
$ terraform version
Terraform v0.12.24
+ provider.aws v2.54.0
```

From what I can tell the code looks absolutely fine, I use this similar pattern a lot in my own terraform - so it's not clear to me why this is occurring.
